cmake: Move declarations up.

hakonfam opened the pull request:
Move several declarations further up in the file.

This is done to make future modifications simpler,
as more variables and functions are available.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

mped-oticon commented:
FYI, this may be superseded by PR https://github.com/zephyrproject-rtos/zephyr/pull/11551.

SebastianBoe commented on CMakeLists.txt:
```diff
@@ -308,6 +308,20 @@ zephyr_cc_option(-Wpointer-arith)
+  set(PRIV_STACK_DEP priv_stacks_prebuilt)
+  endif()
+endif()
+
```
Each independent patch needs to be in it's own commit.

Re-ordering things can break things in unpredictable and subtle ways,
but a clean history can help to debug these issues.

hakonfam replied:
Will try to split it up in more commits.

hakonfam replied:
check it now @SebastianBoe

SebastianBoe replied:
You are missing toup markers, but I guess you know that?

SebastianBoe replied:
LGTM

hakonfam committed:
cmake: Move MPU userspace declaration up.

This is done to make future modifications simpler,
as more variables and functions are available.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam committed:
cmake: Move check for '-nostdinc' up.

This is done to make future modifications simpler,
as more variables and functions are available.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam committed:
cmake: Move check for TOTP property up.

This is done to make future modifications simpler,
as more variables and functions are available.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam committed:
cmake: Move helper function up, inline variable.

This is done to make future modifications simpler,
as more variables and functions are available.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

hakonfam commented:
recheck

carlescufi commented:
@hakonfam can you please rebase?

carlescufi commented:
@nashif can you take a look at this one?

carlescufi commented:
@hakonfam seems like there's an issue with CI. I assume this is a timeout, but @nashif could confirm this.

hakonfam commented:
recheck

nashif reviewed (changes requested):
The CI failures are real, something is not right with the code reordering, bad rebase?
